UnitTests/ControllerTests/CaptionsControllerTest.cs

```diff
@@ -1,316 +1,400 @@
+        public async Task Cancel_Up_Vote_Fail()
+        {
+            var result = await _controller.CancelUpVote("none");
+            Assert.IsType<NotFoundResult>(result.Result);
+
+            result = await _controller.CancelUpVote(null);
+            Assert.IsType<NotFoundResult>(result.Result);
+        }
+
+        [Fact]
+        public async Task Cancel_Down_Vote_Success()
+        {
+            var downvote = 26;
+
+            var caption = new Caption
+            {
+                TranscriptionId = "001",
+                Index = 0,
+                Text = "foo bar",
+                CaptionType = CaptionType.TextCaption,
+                DownVote = downvote
+            };
+
+            _context.Captions.Add(caption);
+            _context.SaveChanges();
+
+            var result = await _controller.CancelDownVote(caption.Id);
+            Assert.Equal(downvote - 1, result.Value.DownVote);
+        }
+
+        [Fact]
+        public async Task Cancel_Down_Vote_Fail()
+        {
+            var result = await _controller.CancelDownVote("none");
+            Assert.IsType<NotFoundResult>(result.Result);
+
+            result = await _controller.CancelDownVote(null);
+            Assert.IsType<NotFoundResult>(result.Result);
+        }
+
+        [Fact]
+        public async Task Search_In_Offering_Fail()
+        {
+            var result = await _controller.SearchInOffering("none", "none");
+            Assert.Empty(result.Value);
+
+            result = await _controller.SearchInOffering("none", null);
+            Assert.Empty(result.Value);
+
+            result = await _controller.SearchInOffering(null, "none");
+            Assert.Empty(result.Value);
+
+            result = await _controller.SearchInOffering(null, null);
+            Assert.Empty(result.Value);
+        }
+
+        [Fact]
+        public async Task Search_In_Offering()
+        {
+            var video = new Video { Id = "789" };
+   
+            var transcriptions = new List<Transcription>()
+            {
+                new Transcription
+                {
+                    Id="1001",
+                    Language="en-US",
+                    VideoId = video.Id
+
+                },
+                new Transcription
+                {
+                    Id="1002",
+                    Language="fr",
+                    VideoId = video.Id
+                }
+            };
+            var captions = new List<Caption>()
+            {
+                new Caption
+                {
+                    Id = "C1",
+                    TranscriptionId = transcriptions[0].Id,
+                    Index = 0,
+                    Text = "hello bob",
+                    CaptionType = CaptionType.TextCaption
+                },
+                new Caption
+                {
+                    Id = "C2",
+                    TranscriptionId = transcriptions[1].Id,
+                    Index = 0,
+                    Text = "hello claude",

```

Comment:
shouldn't this be `bonjour`?

Reply:
haha. Oui. Except the text was chosen to test the language filtering. Next time I'll use a domain-specific language-independent word eg. "Fortran" for test data.